src/acp/policy/removePolicyUrl.ts

```diff
@@ -0,0 +1,59 @@
+/**
+ * Copyright 2021 Inrupt Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal in
+ * the Software without restriction, including without limitation the rights to use,
+ * copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the
+ * Software, and to permit persons to whom the Software is furnished to do so,
+ * subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,
+ * INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A
+ * PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT
+ * HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+ * OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
+ * SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+import type { Url, UrlString } from "../../interfaces";
+import type { WithAccessibleAcr } from "../type/WithAccessibleAcr";
+import { ACP } from "../constants";
+import { getDefaultAccessControlThing } from "../internal/getDefaultAccessControlThing";
+import { DEFAULT_ACCESS_CONTROL } from "../internal/getDefaultAccessControlUrl";
+import { setAccessControlResourceThing } from "../internal/setAccessControlResourceThing";
+import { removeIri } from "../../thing/remove";
+
+/**
+ * ```{note}
+ * The ACP specification is a draft. As such, this function is experimental and
+ * subject to change, even in a non-major release.
+ * See also: https://solid.github.io/authorization-panel/acp-specification/
+ * ```
+ *
+ * Policies allow or deny access modes over resources and their associated
+ * access control resource.
+ *
+ * @param resourceWithAcr The resource for which to remove the URL of a policy
+ * applying to it.
+ * @param policyUrl A Policy URL.
+ * @returns The resource with its ammended access control resource.
+ * @since 1.16.0
+ */
+export function removePolicyUrl<T extends WithAccessibleAcr>(
+  resourceWithAcr: T,
+  policyUrl: Url | UrlString
+): T {
+  const defaultAccessControlThing = getDefaultAccessControlThing(
+    resourceWithAcr,
+    DEFAULT_ACCESS_CONTROL
+  );
+
+  return setAccessControlResourceThing(
+    resourceWithAcr,
+    removeIri(defaultAccessControlThing, ACP.apply, policyUrl)
+  );
+}
```

Comment:
This type of pattern appears often in this PR. Do you think it would make sense eventually to have a layer of functions that only deal with ACRs to add/get/remove policies, and a layer of functions on top of it that accepts input `T extends WithAccessibleAcr` ? It's not something necessary right away, just thinking about how to architecture the code eventually.

Reply:
I think eventually, we need to add the missing layer of Access Controls and cleanup the API to be consistently treating all the RDF things in ACP: Access Control Resource -> Access Control -> Policy -> Matcher.

So yes:
- either we should deal with all the ACP elements as Things consistently if it's the design of the API,
- or we should deal with Things fed directly to the ACRs